scripts/enforce_match_arg_in_assert_produces_warning.py

```diff
@@ -0,0 +1,83 @@
+"""
+
+Enforce that all usages of tm.assert_produces_warning use
+the "match" argument. This will help ensure that users always see
+the correct warning message.
+
+tm.assert_produces_warning(None), tm.assert_produces_warning()
+and tm.assert_produces_warning(False) are excluded as no warning is
+produced.
+
+"""
```

Comment:
Can you remove the extra lines at the start and end?

Reply:
done